internal/pkg/runtime/engine/singularity/cleanup_linux.go

```diff
@@ -63,27 +66,20 @@ func (e *EngineOperations) CleanupContainer(fatal error, status syscall.WaitStat
 	}
 
 	if e.EngineConfig.CryptDev != "" {
```

Comment:
Can you move the whole block before instance block otherwise when run as instance, crypt setup won't be cleanup correctly